### PR TITLE
`auro-formkit`: Add Turborepo & Convert to Monorepo

### PR DESCRIPTION
**Pull request by chrisfalaska**
# `auro-formkit` Monorepo

Creates a new `auro-formkit` repository, powered by [Turborepo](https://turbo.build/repo/docs) and transforms `auro-form` into a monorepo.

## Summary

- Convert the project to a monorepo using Turborepo, reorganizing components and packages under a unified structure.

- Update build scripts and dependencies to support the new architecture.

- Introduce TypeScript support for enhanced development experience.

- Initial documentation added (will further be developed in #11 ).

New Features:
- Introduce Turborepo to manage the monorepo structure, allowing for efficient handling of multiple packages within the repository.

Enhancements:
- Convert the project to a monorepo structure, organizing components and packages under a unified repository.
- Update package dependencies and scripts to align with the new monorepo structure, ensuring compatibility and streamlined operations.
- Enable TypeScript support across various components, enhancing type safety and developer experience.

Build:
- Add a new `turbo.json` configuration file to manage build tasks and dependencies across the monorepo.
- Introduce shared build tools and configurations for consistent build processes across components.

### Install

```
npm i
```

### Build

```
npm run build
```

### Dev

```
npm run dev
```

### Lint

```
npm run lint
```

### Testing

```
npm run test
```

For this PR, `test` is currently a work in progress and is failing.

Further development will continue in https://github.com/AlaskaAirlines/auro-form/issues/85

### CODE REVIEW

File: components/menu/package.json [hunk 1]
```diff
@@ -1,24 +1,16 @@
...
   },
-  "type": "module",
-  "main": "index.js",
-  "license": "Apache-2.0",
-  "engines": {
-    "node": "^18.x || ^20.x"
-  },
   "dependencies": {
     "@alaskaairux/icons": "^4.44.1",
     "@aurodesignsystem/auro-icon": "^6.0.2",
```
Comment (sun-mota) [inline components/menu/package.json:16]: @aurodesignsystem/auro-library version in this package.json is not aligned with the root's package.json

File: components/radio/package.json [hunk 2]
```diff
@@ -1,24 +1,16 @@
...
   },
-  "type": "module",
-  "main": "index.js",
-  "license": "Apache-2.0",
-  "engines": {
-    "node": "^18 || ^20"
-  },
   "dependencies": {
     "@aurodesignsystem/auro-formvalidation": "^1.0.4",
     "@aurodesignsystem/auro-library": "~2.8.0",
```
Comment (sun-mota) [inline components/radio/package.json:16]: @aurodesignsystem/auro-library version here is also not aligned

**Comment by chrisfalaska** [comment]
> 1. Please fix husky pre-commit to remove test, linters should be changed to lint
>    We can add back test to the pre-commit, on the PR for fixing test,
> 2. I don't see dist folder under root. was it intentional?

Changes made here:
https://github.com/AlaskaAirlines/auro-form/pull/84/files/3d3a598dbc96ebb0c78cde8cf8c08573b0ab5625..5a8853d959db2120b890bb3b8ac8498480b761e3

I also left a note to return `test` to Husky pre-commit here:
https://github.com/AlaskaAirlines/auro-form/issues/85#issuecomment-2518424882

File: components/menu/package.json (hunk 1 above)
Comment (chrisfalaska) [reply components/menu/package.json:16]: Moved `auro-library` dependency to root `package.json` and removed from component packages.

File: components/radio/package.json (hunk 2 above)
Comment (chrisfalaska) [reply components/radio/package.json]: Moved `auro-library` dependency to root `package.json` and removed from component packages.

**Review by sun-mota — APPROVED** [review]
build, dev and commit work.
Approve with known issues: publish process, test

**Comment by chrisfalaska** [comment]
> 1. Please fix husky pre-commit to remove test, linters should be changed to lint
>    We can add back test to the pre-commit, on the PR for fixing test,
> 2. I don't see dist folder under root. was it intentional?

re: the root `dist` folder, I created a ticket for the NPM/CDN publishing integration: https://github.com/AlaskaAirlines/auro-form/issues/87

**Pull request merged**